ledger/src/blocktree_db.rs

```diff
@@ -588,6 +588,25 @@ impl Database {
     pub fn storage_size(&self) -> Result<u64> {
         Ok(fs_extra::dir::get_size(&self.path)?)
     }
+
+    // Adds a range to delete to the given write batch and returns whether or not the column has reached
+    // its end
+    pub fn delete_range_cf<C>(&self, batch: &mut WriteBatch, from: Slot, to: Slot) -> Result<bool>
+    where
+        C: Column,
+    {
+        let cf = self.cf_handle::<C>();
+        let from_index = C::as_index(from);
+        let to_index = C::as_index(to);
+        let result = batch.delete_range_cf::<C>(cf, from_index, to_index);
+        let max_slot = self
```

Comment:
@pgarg66 I added logic to detect if we've reached the end of the column to allow the caller to stop looping if purge_slots is given a `None` on our upper bound. I forgot to handle this and it was causing tests to loop :D